### PR TITLE
Added class for highlighted row

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**

### CODE REVIEW

File: src/app/visual/grid/grid-visual.component.ts [hunk 1]
```diff
@@ -54,10 +61,15 @@ export class GridVisualComponent {
     this.dataForSimpleGridWithMultiselect = this.performSort(activeSort, this.dataForSimpleGridWithMultiselect);
   }
 
-  public triggerHighlight() {
+  public triggerTextHighlight() {
     this.highlightText = 'e';
   }
 
+  public triggerRowHighlight() {
+    let row = this.gridRef.nativeElement.querySelectorAll(`tbody tr`)[1];
```
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/visual/grid/grid-visual.component.ts]: Interesting...I hadn't thought of this. Seems like a prudent approach! I'll give it a shot.

[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/visual/grid/grid-visual.component.ts]: What would you say to making this based on the `id` instead of the `index`? I'm worried the index might change based on filtering/searching and complicate things.

File: src/app/visual/grid/grid-visual.component.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import {
-  Component
+  Component, Renderer2, ViewChild, ElementRef
```
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/visual/grid/grid-visual.component.ts]: done.

File: src/app/public/modules/grid/grid.component.scss [hunk 3]
```diff
@@ -36,6 +36,11 @@ $sky-grid-resize-handle-width: 5px;
     background-color: $sky-color-gray-01;
   }
 
+  &.sky-grid-row-highlight {
+    border-top: 1px solid $sky-text-color-action-primary; // overrides normal row border
```
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.scss]: Good call. Done.

[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.scss]: Oh, I forgot - there's a little more to this.

The rows have no top-border by default. However, they do all have the dotted row on the bottom. Without this line, the dotted border shows through the inner 3px highlight and looks like an unintentional artifact. Luckily, grids table-border is set to `collapse`, so by simply giving this row a top border of `1px`, it sits on top of the dotted border above it.

tl;dr - it hides the dotted border above it. :)

File: src/app/visual/grid/grid-visual.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/visual/grid/grid-visual.component.ts]: Done. What do you think?

[the remark this answers is not in the document]
Comment (Blackbaud-SteveBrush) [reply src/app/visual/grid/grid-visual.component.ts]: Works, so long as `id` is a required (and documented) requirement of the model.

File: src/app/public/modules/grid/grid.component.ts [hunk 4]
```diff
@@ -499,6 +502,13 @@ export class SkyGridComponent implements OnInit, AfterContentInit, OnChanges, On
     }
   }
 
+  public isRowHighlighted(id: string) {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/grid/grid.component.ts]: One final thought. Our new TSLint file will require all methods return a type, even if the return type is `void`. Might be good to start doing that now.

```
public isRowHighlighted(id: string): boolean
```

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.ts]: Oops, I already knew this was a style guideline. Thanks for catching that.

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/grid/grid.component.ts]: done.
